snow/networking/sender/sender.go

```diff
@@ -34,27 +34,45 @@ func (s *Sender) Context() *snow.Context { return s.ctx }
 
 // GetAcceptedFrontier ...
 func (s *Sender) GetAcceptedFrontier(validatorIDs ids.ShortSet, requestID uint32) {
-	currentDeadline := time.Time{}
+	// TODO this timeout duration won't exactly match the one that gets registered.
+	// I think that's OK, but double check.
+	timeoutDuration := s.timeouts.TimeoutDuration()
+	deadline := time.Now().Add(timeoutDuration)
+
+	// A GetAcceptedFrontier to myself will always fail.
+	if validatorIDs.Contains(s.ctx.NodeID) {
+		validatorIDs.Remove(s.ctx.NodeID)
+		go s.router.GetAcceptedFrontierFailed(s.ctx.NodeID, s.ctx.ChainID, requestID)
+	}
+
+	// Some of the validators in [validatorIDs] may be benched. That is, they've been unresponsive
+	// so we don't even bother sending messages to them. We just have them immediately fail.
 	for validatorID := range validatorIDs {
+		if s.timeouts.IsBenched(validatorID, s.ctx.ChainID) {
+			validatorIDs.Remove(validatorID)
+			// Immediately register a failure. Do so asynchronously to avoid deadlock.
+			go s.router.GetAcceptedFrontierFailed(s.ctx.NodeID, s.ctx.ChainID, requestID)
```

Comment:
We should be registering this failure into the latency calculator as a timeout at the current value. Otherwise we could run into some weird cases where benching people causes us to bench more people.

Reply:
(this is true with all the requests)